pkgs/native_assets_builder/lib/src/package_layout/package_layout.dart

```diff
@@ -88,25 +89,22 @@ class PackageLayout {
   /// Whether a package has native assets is defined by whether it contains
   /// a `hook/build.dart`.
   ///
-  /// For backwards compatibility, a toplevel `build.dart` is also supported.
-  // TODO(https://github.com/dart-lang/native/issues/823): Remove fallback when
-  // everyone has migrated. (Probably once we stop backwards compatibility of
-  // the protocol version pre 1.2.0 on some future version.)
-  late final Future<List<Package>> packagesWithNativeAssets = () async {
+  /// `package:native` itself is excluded.
+  Future<List<Package>> packagesWithAssets(PipelineStep step) async {
     final result = <Package>[];
     for (final package in packageConfig.packages) {
       final packageRoot = package.root;
       if (packageRoot.scheme == 'file') {
         if (await File.fromUri(
-              packageRoot.resolve('hook/').resolve('build.dart'),
+              packageRoot.resolve('hook/').resolve(step.scriptName),
             ).exists() ||
             await File.fromUri(
-              packageRoot.resolve('build.dart'),
+              packageRoot.resolve(step.scriptName),
             ).exists()) {
           result.add(package);
         }
       }
     }
     return result;
-  }();
+  }
```

Comment:
Invoking this multiple times will now lead to multiple executions, please add in caching.